source/manuals/programming-languages/nodejs/index.html.md.erb

```diff
@@ -253,6 +253,75 @@ const callback = filename => (err, data) => {
 };
 ```
 
+#### Use of `async`/`await`
+
+If your version of Node is above 7.10.0 you should use async functions, and the await keyword, to organise your asynchronous code. This will make it easier to read and write.
+
+```js
+// without async/await
+const getUserFromCreds = function (username, password) {
+  return getUserIdFromAuth(username, password)
+    .then(userId => {
+      return getUserFromApi(userId);
+    })
+    .then(user => {
+      return {
+        'id': user.id,
+        'name': user.name,
+        'age': user.ageInYears      
+      };
+    })
+    .catch(e => {
+      console.log(e);
+    });
+}
+
+// with async/await
+const getUserFromCreds = async function (username, password) {
+  try {
+    const userId = await getUserIdFromAuth(username, password);
+    const user = await getUserFromApi(userId);
+
+    return {
+      'id': user.id,
+      'name': user.name,
+      'age': user.ageInYears
+    };
+  } catch(e) {
+    console.log(e);
+  }
+}
+```
+
+The `await` keyword will pause execution until its promise fulfills. This means a sequence of awaited promises can only execute in the order they're declared.
+
+If you have multiple promises, you should try to [instantiate them separately from when you await them](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await#The_downsides_of_asyncawait), otherwise each promise will be forced to wait for the previous one. Example:
+
+```js
+const getUserFromCreds = async function (username, password) {
+  try {
+    // use await to make this promise blocking
+    await userId = getUserIdFromAuth(username, password);
```

Comment:
I don't think this would be valid syntax
```suggestion
    const userId = await getUserIdFromAuth(username, password);
```

Reply:
Good spot, I corrected this in my test files but not here 👍